.github/workflows/gradle.yml

```diff
@@ -16,17 +16,30 @@ on:
 jobs:
   build:
 
-    runs-on: 
-      group: public-gpu
-      labels: linux-gpu-t4
+    runs-on: ubuntu-latest
     container:
       image: beehivelab/tornadovm-nvidia-openjdk:1.0.6
     permissions:
       contents: read
 
+    env:
+      # The base OS for beehivelab/tornadovm-nvidia-openjdk:1.0.6 is too old and doesn't support Node.js v20 (which depends on glibc 2.28)
+      # As a workaround, we need to both:
+      # * Downgrade Actions to versions that still use node16
+      # * Set ACTIONS_ALLOW_USE_UNSECURE_NODE_VERSION=true so that the runner doesn't force Actions to use node20
+      # See:
+      # * https://github.blog/changelog/2023-09-22-github-actions-transitioning-from-node-16-to-node-20/
+      # * https://github.blog/changelog/2024-05-17-updated-dates-for-actions-runner-using-node20-instead-of-node16-by-default/
+      # * https://github.com/nodejs/node/issues/43246
+      ACTIONS_ALLOW_USE_UNSECURE_NODE_VERSION: true
+
     steps:
+    # We can't use actions/checkout@v4 b/c it uses node20
+    - uses: actions/checkout@v3
```

Comment:
I added `actions/checkout` - you need it for each job that needs the contents of the repo.

I was able to get `./gradlew` to run w/o `actions/setup-java`, but if you need that, make sure you use `actions/setup-java@v3`, since v4 uses upgraded to node20